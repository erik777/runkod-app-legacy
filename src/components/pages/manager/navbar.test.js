import React from 'react';
import NavBar from './navbar';
import TestRenderer from 'react-test-renderer';


const fnProps = {
  logout: () => {
  },
  selectProject: () => {
  },
  history: {
    push: () => {
    }
  },
  toggleUiProp: () => {

  }
};


test('1- default render', () => {
  const props = Object.assign({}, {
    user: {
      username: 'runkod.id.blockstack',
      image: 'https://foo.bar/image.png'
    },
    projects: {
      loading: false
    },
    files: {
      loading: false
    },
    ui: {
      contact: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <NavBar {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('2- no user image', () => {
  const props = Object.assign({}, {
    user: {
      username: 'runkod.id.blockstack',
      image: ''
    },
    projects: {
      loading: false
    },
    files: {
      loading: false
    },
    ui: {
      contact: false
    }

  }, fnProps);

  const renderer = TestRenderer.create(
    <NavBar {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('3- loading', () => {
  const props = Object.assign({}, {
    user: {
      username: 'runkod.id.blockstack',
      image: 'https://foo.bar/image.png'
    },
    projects: {
      loading: true
    },
    files: {
      loading: false
    },
    ui: {
      contact: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <NavBar {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});