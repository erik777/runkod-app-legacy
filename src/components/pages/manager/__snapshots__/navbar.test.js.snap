// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default render 1`] = `
<div
  className="nav-bar"
>
  <div
    className="logo"
    onClick={[Function]}
  >
    <img
      alt="Logo"
      src="logo-white.png"
    />
  </div>
  <div
    className="user-menu"
  >
    <div
      className="user-image"
      style={
        Object {
          "backgroundImage": "url('https://foo.bar/image.png')",
        }
      }
    >
      
    </div>
    <div
      className="menu-list"
    >
      <span
        className="username"
      >
        @
        runkod.id.blockstack
      </span>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Help
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm61 356l-12.2-39.6c13-5.8 24.9-14 35.3-24.4 10.4-10.4 18.6-22.3 24.4-35.3l39.5 12.1c-7.9 19.3-19.7 37-34.9 52.2-15.1 15.3-32.8 27.1-52.1 35zM195 108l12.2 39.6c-13 5.8-24.9 14-35.3 24.4-10.4 10.4-18.6 22.3-24.4 35.3L108 195.2c7.9-19.3 19.7-37 34.9-52.2 15.1-15.3 32.8-27.1 52.1-35zm61 84c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm113.1-49.1c15.2 15.2 26.9 32.9 34.9 52.1l-39.5 12.2c-5.9-13-14-24.9-24.4-35.3-10.4-10.4-22.3-18.6-35.3-24.4l12.1-39.5c19.3 7.9 37 19.7 52.2 34.9zM142.9 369.1c-15.2-15.1-27-32.8-34.9-52.1l39.5-12.2c5.9 13 14 24.9 24.4 35.3 10.4 10.4 22.3 18.6 35.3 24.4L195.1 404c-19.3-7.9-37-19.7-52.2-34.9z"
          />
        </svg>
      </a>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Logout
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 277.4h189.7l-43.6 44.7L368 352l96-96-96-96-31 29.9 44.7 44.7H192v42.8z"
          />
          <path
            d="M255.7 421.3c-44.1 0-85.5-17.2-116.7-48.4-31.2-31.2-48.3-72.7-48.3-116.9 0-44.1 17.2-85.7 48.3-116.9 31.2-31.2 72.6-48.4 116.7-48.4 44 0 85.3 17.1 116.5 48.2l30.3-30.3c-8.5-8.4-17.8-16.2-27.7-23.2C339.7 61 298.6 48 255.7 48 141.2 48 48 141.3 48 256s93.2 208 207.7 208c42.9 0 84-13 119-37.5 10-7 19.2-14.7 27.7-23.2l-30.2-30.2c-31.1 31.1-72.5 48.2-116.5 48.2zM448.004 256.847l-.849-.848.849-.849.848.849z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`2- no user image 1`] = `
<div
  className="nav-bar"
>
  <div
    className="logo"
    onClick={[Function]}
  >
    <img
      alt="Logo"
      src="logo-white.png"
    />
  </div>
  <div
    className="user-menu"
  >
    <div
      className="user-image"
      style={Object {}}
    >
      R
    </div>
    <div
      className="menu-list"
    >
      <span
        className="username"
      >
        @
        runkod.id.blockstack
      </span>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Help
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm61 356l-12.2-39.6c13-5.8 24.9-14 35.3-24.4 10.4-10.4 18.6-22.3 24.4-35.3l39.5 12.1c-7.9 19.3-19.7 37-34.9 52.2-15.1 15.3-32.8 27.1-52.1 35zM195 108l12.2 39.6c-13 5.8-24.9 14-35.3 24.4-10.4 10.4-18.6 22.3-24.4 35.3L108 195.2c7.9-19.3 19.7-37 34.9-52.2 15.1-15.3 32.8-27.1 52.1-35zm61 84c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm113.1-49.1c15.2 15.2 26.9 32.9 34.9 52.1l-39.5 12.2c-5.9-13-14-24.9-24.4-35.3-10.4-10.4-22.3-18.6-35.3-24.4l12.1-39.5c19.3 7.9 37 19.7 52.2 34.9zM142.9 369.1c-15.2-15.1-27-32.8-34.9-52.1l39.5-12.2c5.9 13 14 24.9 24.4 35.3 10.4 10.4 22.3 18.6 35.3 24.4L195.1 404c-19.3-7.9-37-19.7-52.2-34.9z"
          />
        </svg>
      </a>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Logout
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 277.4h189.7l-43.6 44.7L368 352l96-96-96-96-31 29.9 44.7 44.7H192v42.8z"
          />
          <path
            d="M255.7 421.3c-44.1 0-85.5-17.2-116.7-48.4-31.2-31.2-48.3-72.7-48.3-116.9 0-44.1 17.2-85.7 48.3-116.9 31.2-31.2 72.6-48.4 116.7-48.4 44 0 85.3 17.1 116.5 48.2l30.3-30.3c-8.5-8.4-17.8-16.2-27.7-23.2C339.7 61 298.6 48 255.7 48 141.2 48 48 141.3 48 256s93.2 208 207.7 208c42.9 0 84-13 119-37.5 10-7 19.2-14.7 27.7-23.2l-30.2-30.2c-31.1 31.1-72.5 48.2-116.5 48.2zM448.004 256.847l-.849-.848.849-.849.848.849z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`3- loading 1`] = `
<div
  className="nav-bar loading"
>
  <div
    className="logo"
    onClick={[Function]}
  >
    <img
      alt="Logo"
      src="logo-white.png"
    />
  </div>
  <div
    className="user-menu"
  >
    <div
      className="user-image"
      style={
        Object {
          "backgroundImage": "url('https://foo.bar/image.png')",
        }
      }
    >
      
    </div>
    <div
      className="menu-list"
    >
      <span
        className="username"
      >
        @
        runkod.id.blockstack
      </span>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Help
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm61 356l-12.2-39.6c13-5.8 24.9-14 35.3-24.4 10.4-10.4 18.6-22.3 24.4-35.3l39.5 12.1c-7.9 19.3-19.7 37-34.9 52.2-15.1 15.3-32.8 27.1-52.1 35zM195 108l12.2 39.6c-13 5.8-24.9 14-35.3 24.4-10.4 10.4-18.6 22.3-24.4 35.3L108 195.2c7.9-19.3 19.7-37 34.9-52.2 15.1-15.3 32.8-27.1 52.1-35zm61 84c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm113.1-49.1c15.2 15.2 26.9 32.9 34.9 52.1l-39.5 12.2c-5.9-13-14-24.9-24.4-35.3-10.4-10.4-22.3-18.6-35.3-24.4l12.1-39.5c19.3 7.9 37 19.7 52.2 34.9zM142.9 369.1c-15.2-15.1-27-32.8-34.9-52.1l39.5-12.2c5.9 13 14 24.9 24.4 35.3 10.4 10.4 22.3 18.6 35.3 24.4L195.1 404c-19.3-7.9-37-19.7-52.2-34.9z"
          />
        </svg>
      </a>
      <a
        className="menu-list-item"
        onClick={[Function]}
      >
        Logout
         
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 277.4h189.7l-43.6 44.7L368 352l96-96-96-96-31 29.9 44.7 44.7H192v42.8z"
          />
          <path
            d="M255.7 421.3c-44.1 0-85.5-17.2-116.7-48.4-31.2-31.2-48.3-72.7-48.3-116.9 0-44.1 17.2-85.7 48.3-116.9 31.2-31.2 72.6-48.4 116.7-48.4 44 0 85.3 17.1 116.5 48.2l30.3-30.3c-8.5-8.4-17.8-16.2-27.7-23.2C339.7 61 298.6 48 255.7 48 141.2 48 48 141.3 48 256s93.2 208 207.7 208c42.9 0 84-13 119-37.5 10-7 19.2-14.7 27.7-23.2l-30.2-30.2c-31.1 31.1-72.5 48.2-116.5 48.2zM448.004 256.847l-.849-.848.849-.849.848.849z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
